src/Optimization/hiopDualsUpdater.cpp

```diff
@@ -413,39 +413,63 @@ bool hiopDualsLsqUpdateLinsysAugSparse::do_lsq_update(hiopIterate& iter,
         
 #endif  // HIOP_USE_STRUMPACK
       }
-    } //KS: /end of CPU mode/ do not put CU SOLVER anywhere above this!!!!!
-else {
+      //KS: /end of CPU mode/ do not put CU SOLVER anywhere above this!!!!!
+    } else {  
       //
-      // we're on device
+      // We're on device
       //
-#ifdef HIOP_USE_CUSOLVER // This is our first choice on the device!
+      // Under hybrid compute_mode, LSQ-based initialization can be done using CPU sparse linear solvers.
+      // Under gpu compute_mode, which is work in progress, the initialization should be done only using
+      // GPU sparse linear solvers.
+      
+#ifdef HIOP_USE_CUSOLVER 
+      if(compute_mode == "gpu") {
         assert((linear_solver == "cusolver" || linear_solver == "auto") &&
                "the value for duals_init_linear_solver_sparse is invalid and should have been corrected during "
                "options processing");
-               hiopLinSolverIndefSparseCUSOLVER *p = new hiopLinSolverIndefSparseCUSOLVER(n, nnz, nlp_);
+      }
+
+      // This is our first choice on the device.
+      if(linear_solver == "cusolver" || linear_solver == "auto") {
+        hiopLinSolverIndefSparseCUSOLVER *p = new hiopLinSolverIndefSparseCUSOLVER(n, nnz, nlp_);
         nlp_->log->printf(hovSummary,
                           "LSQ Dual Initialization --- KKT_SPARSE_XDYcYd linsys: using CUSOLVER on device as an "
                           "indefinite solver, size %d (%d cons)\n",
                           n,
                           neq+nineq);
         p->setFakeInertia(neq + nineq);
```

Comment:
Remove fake inertia from here. It is not used anyways.
```suggestion
      if(linear_solver == "cusolver" || linear_solver == "auto") {
        hiopLinSolverIndefSparseCUSOLVER *p = new hiopLinSolverIndefSparseCUSOLVER(n, nnz, nlp_);
        nlp_->log->printf(hovSummary,
                          "LSQ Dual Initialization --- KKT_SPARSE_XDYcYd linsys: using CUSOLVER on device as an "
                          "indefinite solver, size %d (%d cons)\n",
                          n,
                          neq+nineq);
```

Reply:
it turns out that cuSOLVER LU still relies on the fake inertia in this branch and stops working if I remove it.